### PR TITLE
Add CI tests and image for hw-event-proxy-operator

### PR DESCRIPTION
**Pull request by jzding**
Signed-off-by: Jack Ding <jacding@redhat.com>

### CODE REVIEW

**Comment by jzding** [comment]
ci/rehearse/redhat-cne/hw-event-proxy-operator/main/e2e-operator

File: core-services/prow/02_config/_plugins.yaml [hunk 1]
```diff
@@ -256,27 +256,6 @@ cherry_pick_unapproved:
...
-    build_farm_ocp:
-      clusters:
-      - app.ci
-      - build01
-      - build02
-      - build03
-      - build04
-      - vsphere
-      namespaces:
-      - ocp
```
Comment (bparees) [inline core-services/prow/02_config/_plugins.yaml:279]: the changes to this file don't seem related to this PR, are they deliberate?  was their a tooling change and this is just the first PR since the tooling change?

@petr-muller

File: core-services/image-mirroring/openshift/mapping_origin_4_10 [hunk 2]
```diff
@@ -27,6 +27,7 @@ registry.ci.openshift.org/origin/4.10:azure-file-csi-driver-operator quay.io/ope
 registry.ci.openshift.org/origin/4.10:azure-file-csi-driver-operator-test quay.io/openshift/origin-azure-file-csi-driver-operator-test:4.10 quay.io/openshift/origin-azure-file-csi-driver-operator-test:4.10.0 quay.io/openshift/origin-azure-file-csi-driver-operator-test:latest
 registry.ci.openshift.org/origin/4.10:azure-machine-controllers quay.io/openshift/origin-azure-machine-controllers:4.10 quay.io/openshift/origin-azure-machine-controllers:4.10.0 quay.io/openshift/origin-azure-machine-controllers:latest
 registry.ci.openshift.org/origin/4.10:baremetal-hardware-event-proxy quay.io/openshift/origin-baremetal-hardware-event-proxy:4.10 quay.io/openshift/origin-baremetal-hardware-event-proxy:4.10.0 quay.io/openshift/origin-baremetal-hardware-event-proxy:latest
+registry.ci.openshift.org/origin/4.10:baremetal-hardware-event-proxy-operator quay.io/openshift/origin-baremetal-hardware-event-proxy-operator:4.10 quay.io/openshift/origin-baremetal-hardware-event-proxy-operator:4.10.0 quay.io/openshift/origin-baremetal-hardware-event-proxy-operator:latest
```
Comment (bparees) [inline core-services/image-mirroring/openshift/mapping_origin_4_10:30]: the name looks fine, so i'm prepared to approve that part of this PR.

File: core-services/prow/02_config/_plugins.yaml (hunk 1 above)
Comment (petr-muller) [reply core-services/prow/02_config/_plugins.yaml:279]: I am not aware of any tool changes that would cause this. I agree that I would not expect this change to be there. How was this PR created?

**Review by petr-muller — COMMENTED** [review]
/hold

This probably also needs `_prowconfig.yaml` file in `redhat-cne/hw-event-proxy-operator/` to configure at leeast Tide merge criteria.

File: core-services/prow/02_config/_plugins.yaml (hunk 1 above)
Comment (jzding) [reply core-services/prow/02_config/_plugins.yaml:279]: This PR was created by `make new-repo`. I did not manually make any changes under `core-services/prow/02_config`. They were automatically done by the script.

Comment (jzding) [reply core-services/prow/02_config/_plugins.yaml:279]: I can try remove changes under `core-services/prow/02_config` and see if tests can still pass.

Comment (petr-muller) [reply core-services/prow/02_config/_plugins.yaml:279]: Thanks, I'll try to reproduce and file that as a bug. Please remove the changes in this file from this PR.

Comment (petr-muller) [reply core-services/prow/02_config/_plugins.yaml]: https://issues.redhat.com/browse/DPTP-2721

Comment (petr-muller) [reply core-services/prow/02_config/_plugins.yaml]: Test will not pass if you do that. The unwanted change is just in `core-services/prow/02_config/_plugins.yaml` file, the `_pluginconfig.yaml` that was originally present in a directory structure under `core-services/prow/02_config` was actually needed.

Comment (jzding) [reply core-services/prow/02_config/_plugins.yaml]: Done. Removed just the `core-services/prow/02_config/_plugins.yaml` as suggested.

**Comment by jzding** [comment]
> /hold
> 
> This probably also needs `_prowconfig.yaml` file in `redhat-cne/hw-event-proxy-operator/` to configure at leeast Tide merge criteria.

Added _prowconfig.yaml

**Comment by bparees** [comment]
approving mirror/naming bits:

/approve

I didn't see any other issues, but i didn't analyze your test/prow configs in detail.  Since this is an olm operator and not in payload, i view that as largely the purview of the owning team.

**Review by petr-muller — APPROVED** [review]

**Comment by jzding** [comment]
@petr-muller could you delete the /hold label?

**Comment by petr-muller** [comment]
/hold cancel

@jzding you can do ^^^ as well
